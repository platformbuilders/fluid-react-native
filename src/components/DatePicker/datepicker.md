
# DatePicker ðŸ“…

```js
import React from 'react';
import { DatePicker } from '@platformbuilders/react-native-ui';

const Home: React.FC = () => {
  return (
      <DatePicker
        mode='date'
        androidMode='spinner'
        locale='pt-BR'
        format='DD/MM/YYYY'
      />
  );
}

export default Home;
```


| Name  | Required | Type | Default Value | Description 
| ------------- | ------------- | ------------- |------------- |------------- |
| value | false | string | ' ' | value datePicker ex: 2020-25-12 |
| label | false | string | ' ' |  |
| dark | false | boolean | false | style dark  |
| editable | false | boolean | true |  |
| cancelBtnText | false | string | Cancelar |  |
| confirmBtnText | false | string | Confirmar |  |
| testID | false | string | ' ' |  |
| accessibilityLabel | false | string | ' ' |  |
| mode | true | string | date |  |
| androidMode | true | string | spinner |  |
| locale | true | string | pt-BR |  |
| onDateChange | false | (x: string): void | (): void => {} |  |
| maxDate | false | string | |  |
| format | true | string | 'DD/MM/YYYY' |  |



