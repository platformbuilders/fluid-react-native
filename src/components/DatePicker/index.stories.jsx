import { storiesOf } from '@storybook/react-native';
import datepickerMarkdown from './datepicker.md';

storiesOf('DatePicker', module)
  .addParameters({
    readme: {
      content: datepickerMarkdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => {});
