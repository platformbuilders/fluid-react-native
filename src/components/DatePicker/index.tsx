import React, { FC, useState, useCallback, useEffect } from 'react';
import { Animated } from 'react-native';
import { DatePickerProps } from 'react-native-datepicker';
import FormError from '../FormError';
import {
  Label,
  BottomLine,
  DatePicker,
  DatePickerStyles,
  DatePickerStylesDark,
  LABEL_UPPER_STYLE,
  LABEL_LOWER_STYLE,
} from './styles';

interface Props {
  value?: string;
  label?: string;
  error?: string;
  cancelBtnText?: string;
  testID?: string;
  accessibilityLabel?: string;
  confirmBtnText?: string;
  mode?: DatePickerProps['mode'];
  androidMode?: DatePickerProps['androidMode'];
  onDateChange?(x: string): void;
  maxDate?: string;
  editable?: boolean;
  locale?: string;
  format?: string;
  dark?: boolean;
  status?: string;
  id?: string;
  accessibility: string;
}

const DatePickerInput: FC<Props> = ({
  id,
  accessibility,
  label = '',
  error = '',
  dark = false,
  editable = true,
  value = '',
  testID,
  mode = 'date',
  androidMode = 'spinner',
  locale = 'pt-BR',
  format = 'DD/MM/YYYY',
  cancelBtnText = 'Cancelar',
  confirmBtnText = 'Confirmar',
  onDateChange = (): void => {},
  maxDate,
}) => {
  const [labelAnimatedStyle] = useState({
    top: new Animated.Value(LABEL_LOWER_STYLE.top),
    fontSize: new Animated.Value(LABEL_LOWER_STYLE.fontSize),
  });

  const [date, setDate] = useState('');
  const [isPlaceholder, setIsPlaceHolder] = useState(true);

  const execAnimation = useCallback(() => {
    const animations = Object.keys(LABEL_UPPER_STYLE).map((animationProp) =>
      Animated.timing(labelAnimatedStyle[animationProp], {
        toValue: LABEL_UPPER_STYLE[animationProp],
        duration: 200,
        useNativeDriver: false,
      }),
    );

    setIsPlaceHolder(false);

    Animated.parallel(animations).start();
  }, [Animated, labelAnimatedStyle]);

  const updateDate = useCallback(
    (updatedDate: string): void => {
      setDate(updatedDate);
      execAnimation();
      if (onDateChange) {
        onDateChange(updatedDate);
      }
    },
    [onDateChange],
  );

  useEffect(() => {
    if (value) {
      execAnimation();
    }
  }, [value]);

  const customStyles = error
    ? {
        ...DatePickerStyles,
        dateInput: { borderColor: '#cc0000' },
      }
    : dark
    ? DatePickerStylesDark
    : DatePickerStyles;

  return (
    <FormError
      id={id || accessibility}
      accessibility={accessibility}
      error={error}
    >
      <Label
        error={error || ''}
        style={labelAnimatedStyle}
        isPlaceholder={isPlaceholder}
        dark={dark}
      >
        {label}
      </Label>
      <DatePicker
        mode={mode}
        androidMode={androidMode}
        locale={locale}
        placeholder=" "
        format={format}
        editable={!editable}
        date={date}
        customStyles={customStyles}
        maxDate={maxDate}
        testID={testID || accessibility}
        confirmBtnText={confirmBtnText}
        cancelBtnText={cancelBtnText}
        onDateChange={updateDate}
        showIcon={false}
        dark={dark}
      />
      <BottomLine dark={dark} />
    </FormError>
  );
};

export default DatePickerInput;
