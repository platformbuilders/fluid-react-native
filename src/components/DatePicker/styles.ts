import { Animated } from 'react-native';
import DefaultDatePicker, {
  DatePickerCustomStylesProps,
} from 'react-native-datepicker';
import styled from 'styled-components/native';
import { ComponentType } from 'react';
import { getTheme, switchStyle } from '../../utils/helpers';
import { InputStatus } from '../../types';

const getStatusStyle = switchStyle('status');
const primaryContrast = getTheme('primary.contrast');
const primaryDark = getTheme('primary.dark');
const disabled = getTheme('disabled.main');
const inputMainColor = (props: { dark: any }): any =>
  getStatusStyle({
    [InputStatus.Success]: getTheme('success'),
    [InputStatus.Failure]: getTheme('failure'),
    [InputStatus.Default]: props.dark ? primaryDark : primaryContrast,
    [InputStatus.Disabled]: disabled,
  });

export const LABEL_UPPER_STYLE = {
  top: 8,
  fontSize: 14,
};

export const LABEL_LOWER_STYLE = {
  top: 40,
  fontSize: 18,
};

interface TextProps {
  dark?: boolean;
  error?: string;
  isPlaceholder?: boolean;
}

export const TextLabel = styled.Text<TextProps>`
  line-height: 19px;
  padding-bottom: 15px;
  color: ${(props: TextProps): string =>
    props.dark ? primaryDark(props) : primaryContrast(props)};
`;

export const Label = Animated.createAnimatedComponent<ComponentType<any>>(
  TextLabel,
);

interface DatePickerProps {
  customStyles?: DatePickerCustomStylesProps;
  dark?: boolean;
  editable: boolean;
}
export const DatePicker = styled(DefaultDatePicker)<DatePickerProps>`
  width: 100%;
`;

export const BottomLine = styled.View`
  height: 1px;
  background-color: ${inputMainColor};
`;

const commonDatePickerStyles = {
  disabled: {
    backgroundColor: 'transparent',
  },
  dateInput: {
    borderWidth: 0,
  },
  dateTouchBody: {
    width: '100%',
  },
};

export const DatePickerStyles: DatePickerCustomStylesProps = {
  ...commonDatePickerStyles,
  placeholderText: {
    position: 'absolute',
    left: 0,
    color: '#fff',
    fontSize: LABEL_LOWER_STYLE.fontSize,
  },
  dateText: {
    position: 'absolute',
    left: 0,
    color: '#fff',
    fontSize: LABEL_UPPER_STYLE.fontSize,
    fontWeight: '700',
  },
};

export const DatePickerStylesDark: DatePickerCustomStylesProps = {
  ...commonDatePickerStyles,
  placeholderText: {
    position: 'absolute',
    left: 0,
    color: '#000',
    fontSize: LABEL_LOWER_STYLE.fontSize,
  },
  dateText: {
    position: 'absolute',
    left: 0,
    color: '#000',
    fontSize: LABEL_UPPER_STYLE.fontSize,
    fontWeight: '700',
  },
};
