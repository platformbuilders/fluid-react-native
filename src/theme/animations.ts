import { Animations } from '@platformbuilders/theme-toolkit';
import {
  LoadDotsAnimation,
  LoadDotsContrastAnimation,
  LoadingCircleAnimation,
  LoadingCircleContrastAnimation,
  LoadingProgressAnimation,
} from '../assets/animations';

export default {
  circularLoading: LoadingCircleAnimation,
  contrastCircularLoading: LoadingCircleContrastAnimation,
  buttonLoading: LoadDotsAnimation,
  contrastButtonLoading: LoadDotsContrastAnimation,
  linearLoading: LoadingProgressAnimation,
  contrastLinearLoading: LoadingProgressAnimation,
} as Animations;
