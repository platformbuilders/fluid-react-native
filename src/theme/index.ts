export { default as colors } from './colors';
export { default as spacing } from './spacing';
export { default as radius } from './radius';
export { default as animations } from './animations';
export { default as typography } from './typography';
