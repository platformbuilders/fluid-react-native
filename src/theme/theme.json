{
  "spacing": {
    "min": {
      "value": 1,
      "type": "spacing"
    },
    "xxs": {
      "value": 2,
      "type": "spacing"
    },
    "xs": {
      "value": 4,
      "type": "spacing"
    },
    "sm": {
      "value": 8,
      "type": "spacing"
    },
    "md": {
      "value": 16,
      "type": "spacing"
    },
    "lg": {
      "value": 24,
      "type": "spacing"
    },
    "xl": {
      "value": 32,
      "type": "spacing"
    },
    "xxl": {
      "value": 48,
      "type": "spacing"
    },
    "max": {
      "value": 64,
      "type": "spacing"
    }
  },
  "sizing": {
    "min": {
      "value": 1,
      "type": "sizing"
    },
    "xxs": {
      "value": 2,
      "type": "sizing"
    },
    "xs": {
      "value": 4,
      "type": "sizing"
    },
    "sm": {
      "value": 8,
      "type": "sizing"
    },
    "md": {
      "value": 16,
      "type": "sizing"
    },
    "lg": {
      "value": 24,
      "type": "sizing"
    },
    "xl": {
      "value": 32,
      "type": "sizing"
    },
    "xxl": {
      "value": 48,
      "type": "sizing"
    },
    "max": {
      "value": 64,
      "type": "sizing"
    }
  },
  "borderRadius": {
    "min": {
      "value": 0,
      "type": "borderRadius"
    },
    "xxs": {
      "value": 1,
      "type": "borderRadius"
    },
    "xs": {
      "value": 2,
      "type": "borderRadius"
    },
    "sm": {
      "value": 4,
      "type": "borderRadius"
    },
    "md": {
      "value": 6,
      "type": "borderRadius"
    },
    "lg": {
      "value": 8,
      "type": "borderRadius"
    },
    "xl": {
      "value": 16,
      "type": "borderRadius"
    },
    "xxl": {
      "value": 32,
      "type": "borderRadius"
    },
    "max": {
      "value": 999,
      "type": "borderRadius"
    }
  },
  "borderWidth": {
    "min": {
      "value": 0.5,
      "type": "borderWidth"
    },
    "xxs": {
      "value": 1,
      "type": "borderWidth"
    },
    "xs": {
      "value": 1.5,
      "type": "borderWidth"
    },
    "sm": {
      "value": 2,
      "type": "borderWidth"
    },
    "md": {
      "value": 3,
      "type": "borderWidth"
    },
    "lg": {
      "value": 4,
      "type": "borderWidth"
    },
    "xl": {
      "value": 6,
      "type": "borderWidth"
    },
    "xxl": {
      "value": 8,
      "type": "borderWidth"
    },
    "max": {
      "value": 12,
      "type": "borderWidth"
    }
  },
  "boxShadow": {
    "z0": {
      "value": {
        "x": 0,
        "y": 0,
        "spread": 0,
        "color": "",
        "blur": 0
      },
      "type": "boxShadow"
    },
    "z1": {
      "value": {
        "x": 0,
        "y": 1,
        "spread": 0,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 3
      },
      "type": "boxShadow"
    },
    "z2": {
      "value": {
        "x": 0,
        "y": 4,
        "spread": -1,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 6
      },
      "type": "boxShadow"
    },
    "z3": {
      "value": {
        "x": 0,
        "y": 10,
        "spread": -3,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 15
      },
      "type": "boxShadow"
    },
    "z4": {
      "value": {
        "x": 0,
        "y": 20,
        "spread": -5,
        "color": "rgba(0, 0, 0, 0.1)",
        "blur": 25
      },
      "type": "boxShadow"
    },
    "z5": {
      "value": {
        "x": 0,
        "y": 25,
        "spread": -12,
        "color": "rgba(0, 0, 0, 0.2)",
        "blur": 50
      },
      "type": "boxShadow"
    }
  },
  "brand": {
    "primary": {
      "light": {
        "value": "#7684F9",
        "type": "color"
      },
      "main": {
        "value": "#4255EF",
        "type": "color"
      },
      "dark": {
        "value": "#1B288E",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "secondary": {
      "light": {
        "value": "#8BD6F8",
        "type": "color"
      },
      "main": {
        "value": "#42ADEF",
        "type": "color"
      },
      "dark": {
        "value": "#337BB7",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "tertiary": {
      "light": {
        "value": "#BA96F5",
        "type": "color"
      },
      "main": {
        "value": "#8442EF",
        "type": "color"
      },
      "dark": {
        "value": "#4A00DC",
        "type": "color"
      },
      "contrast": {
        "value": "#ffffff",
        "type": "color"
      }
    },
    "accent": {
      "light": {
        "value": "#FBD268",
        "type": "color"
      },
      "main": {
        "value": "#F49D19",
        "type": "color"
      },
      "dark": {
        "value": "#A1530A",
        "type": "color"
      },
      "contrast": {
        "value": "#351B02",
        "type": "color"
      }
    }
  },
  "success": {
    "light": {
      "value": "#F4FCD1",
      "type": "color"
    },
    "main": {
      "value": "#96C920",
      "type": "color"
    },
    "dark": {
      "value": "#396006",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "danger": {
    "light": {
      "value": "#FFE8D8",
      "type": "color",
      "description": "the 100 shade of a color"
    },
    "main": {
      "value": "#FF4F3F",
      "type": "color"
    },
    "dark": {
      "value": "#7A0C27",
      "type": "color",
      "description": "the 900s shade of a color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "warning": {
    "light": {
      "value": "#FFF8CE",
      "type": "color"
    },
    "main": {
      "value": "#FFCA0C",
      "type": "color"
    },
    "dark": {
      "value": "#7A5502",
      "type": "color"
    },
    "contrast": {
      "value": "#261B00",
      "type": "color"
    }
  },
  "info": {
    "light": {
      "value": "#C2ECEC",
      "type": "color"
    },
    "main": {
      "value": "#42CCD4",
      "type": "color"
    },
    "dark": {
      "value": "#186F82",
      "type": "color"
    },
    "contrast": {
      "value": "#ffffff",
      "type": "color"
    }
  },
  "text": {
    "main": {
      "value": "#444444",
      "type": "color"
    },
    "dark": {
      "value": "#000",
      "type": "color"
    },
    "light": {
      "value": "#999999",
      "type": "color"
    },
    "contrast": {
      "value": "#fff",
      "type": "color"
    }
  },
  "background": {
    "z0": {
      "value": "#ffffff",
      "type": "color"
    },
    "z1": {
      "value": "#fafafa",
      "type": "color"
    },
    "z2": {
      "value": "#f0f0f0",
      "type": "color"
    },
    "z3": {
      "value": "#dddddd",
      "type": "color"
    },
    "z4": {
      "value": "#d1d1d1",
      "type": "color"
    }
  },
  "opacity": {
    "disabled": {
      "value": "30%",
      "type": "opacity"
    },
    "empty": {
      "value": "50%",
      "type": "opacity"
    },
    "overlay": {
      "value": "80%",
      "type": "opacity"
    }
  },
  "fontFamilies": {
    "heading": {
      "value": "Roboto",
      "type": "fontFamilies"
    },
    "body": {
      "value": "Roboto",
      "type": "fontFamilies"
    }
  },
  "lineHeights": {
    "heading": {
      "value": "110%",
      "type": "lineHeights"
    },
    "body": {
      "value": "140%",
      "type": "lineHeights"
    }
  },
  "letterSpacing": {
    "default": {
      "value": 0,
      "type": "letterSpacing"
    },
    "bigger": {
      "value": "105%",
      "type": "letterSpacing"
    }
  },
  "paragraphSpacing": {
    "h1": {
      "value": 32,
      "type": "paragraphSpacing"
    },
    "h2": {
      "value": 26,
      "type": "paragraphSpacing"
    }
  },
  "fontWeights": {
    "headingRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "headingBold": {
      "value": "Bold",
      "type": "fontWeights"
    },
    "bodyRegular": {
      "value": "Regular",
      "type": "fontWeights"
    },
    "bodyBold": {
      "value": "Bold",
      "type": "fontWeights"
    }
  },
  "fontSizes": {
    "min": {
      "value": 10,
      "type": "fontSizes"
    },
    "xxs": {
      "value": 11,
      "type": "fontSizes"
    },
    "xs": {
      "value": 13,
      "type": "fontSizes"
    },
    "sm": {
      "value": 14,
      "type": "fontSizes"
    },
    "md": {
      "value": 15,
      "type": "fontSizes"
    },
    "lg": {
      "value": 17,
      "type": "fontSizes"
    },
    "xl": {
      "value": 20,
      "type": "fontSizes"
    },
    "xxl": {
      "value": 24,
      "type": "fontSizes"
    },
    "max": {
      "value": 32,
      "type": "fontSizes"
    }
  },
  "themeRadius": {
    "button": {
      "value": 3,
      "type": "borderRadius"
    },
    "card": {
      "value": 6,
      "type": "borderRadius"
    },
    "input": {
      "value": 3,
      "type": "borderRadius"
    }
  },
  "typography": {
    "H1": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h1",
          "paragraphSpacing": "$paragraphSpacing.h1",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "H2": {
      "Bold": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingBold",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      },
      "Regular": {
        "type": "typography",
        "value": {
          "fontFamily": "$fontFamilies.heading",
          "fontWeight": "$fontWeights.headingRegular",
          "lineHeight": "$lineHeights.heading",
          "fontSize": "$fontSizes.h2",
          "paragraphSpacing": "$paragraphSpacing.h2",
          "letterSpacing": "$letterSpacing.decreased"
        }
      }
    },
    "Body": {
      "type": "typography",
      "value": {
        "fontFamily": "$fontFamilies.body",
        "fontWeight": "$fontWeights.bodyRegular",
        "lineHeight": "$lineHeights.heading",
        "fontSize": "$fontSizes.body",
        "paragraphSpacing": "$paragraphSpacing.h2"
      }
    }
  }
}
