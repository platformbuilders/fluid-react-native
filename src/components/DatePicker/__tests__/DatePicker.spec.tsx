import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import DatePicker from '..';
import theme from '../../../theme';

// Mock para o módulo react-native-datepicker usado pelo componente
jest.mock('react-native-datepicker', () => 'DatePicker');

// Mock para módulo Animated
jest.mock('react-native', () => {
  const rn = jest.requireActual('react-native');
  rn.Animated = {
    Value: jest.fn(() => ({
      setValue: jest.fn(),
      interpolate: jest.fn(),
    })),
    timing: jest.fn(() => ({
      start: jest.fn(),
    })),
    parallel: jest.fn(() => ({
      start: jest.fn(),
    })),
    createAnimatedComponent: jest.fn((comp) => comp),
  };
  return rn;
});

jest.useFakeTimers();

describe('<DatePicker />', () => {
  // Definindo constantes para reutilização
  const TEST_ID = 'testing';
  const TEST_VALUE = 'test';
  const TEST_LABEL = 'test';
  const TEST_ERROR = 'test';
  const TEST_BTN_TEXT = 'test';

  it('should render datepicker', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <DatePicker id={TEST_ID} accessibility="" />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render datepicker with value', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <DatePicker id={TEST_ID} accessibility="" value={TEST_VALUE} />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render datepicker with label', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <DatePicker id={TEST_ID} accessibility="" label={TEST_LABEL} />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render datepicker with error', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <DatePicker id={TEST_ID} accessibility="" error={TEST_ERROR} />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render datepicker with button cancel text', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <DatePicker
          id={TEST_ID}
          accessibility=""
          cancelBtnText={TEST_BTN_TEXT}
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render datepicker with custom props', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <DatePicker
          id={TEST_ID}
          accessibility=""
          testID={TEST_VALUE}
          accessibilityLabel="test"
          confirmBtnText="test"
          mode="date"
          androidMode="calendar"
          editable={false}
          locale="en"
          format="yyyy-MM-dd"
          dark={false}
          status="disabled"
          maxDate="2021-12-31"
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should call onDateChange when date is changed', () => {
    const onDateChangeMock = jest.fn();
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <DatePicker
          id={TEST_ID}
          accessibility=""
          onDateChange={onDateChangeMock}
        />
      </ThemeProvider>,
    );

    // O testID no componente é atribuído como accessibility ou o valor de testID
    const datePicker = wrapper.root.findByProps({ testID: '' });

    // Simular mudança de data usando act() para envolver atualizações de estado
    act(() => {
      datePicker.props.onDateChange('2021-01-01');
    });

    // Avançar timers para completar animações
    act(() => {
      jest.advanceTimersByTime(300);
    });

    expect(onDateChangeMock).toHaveBeenCalledWith('2021-01-01');
  });
});
