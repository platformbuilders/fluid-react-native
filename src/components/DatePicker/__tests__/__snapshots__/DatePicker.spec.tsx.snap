// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> should render datepicker 1`] = `
[
  <Text
    accessibilityLabel=""
    collapsable={false}
    dark={false}
    error=""
    isPlaceholder={true}
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#ffffff",
          "lineHeight": 19,
          "paddingBottom": 16,
        },
        {
          "fontSize": 18,
          "top": 40,
        },
      ]
    }
    variant="md"
  />,
  <DatePicker
    androidMode="spinner"
    cancelBtnText="Cancelar"
    confirmBtnText="Confirmar"
    customStyles={
      {
        "dateInput": {
          "borderWidth": 0,
        },
        "dateText": {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "700",
          "left": 0,
          "position": "absolute",
        },
        "dateTouchBody": {
          "width": "100%",
        },
        "disabled": {
          "backgroundColor": "transparent",
        },
        "placeholderText": {
          "color": "#fff",
          "fontSize": 18,
          "left": 0,
          "position": "absolute",
        },
      }
    }
    dark={false}
    date=""
    editable={false}
    format="DD/MM/YYYY"
    locale="pt-BR"
    mode="date"
    onDateChange={[Function]}
    placeholder=" "
    showIcon={false}
    style={
      [
        {
          "width": "100%",
        },
      ]
    }
  />,
  <View
    dark={false}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "height": 0.5,
        },
      ]
    }
  />,
]
`;

exports[`<DatePicker /> should render datepicker with button cancel text 1`] = `
[
  <Text
    accessibilityLabel=""
    collapsable={false}
    dark={false}
    error=""
    isPlaceholder={true}
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#ffffff",
          "lineHeight": 19,
          "paddingBottom": 16,
        },
        {
          "fontSize": 18,
          "top": 40,
        },
      ]
    }
    variant="md"
  />,
  <DatePicker
    androidMode="spinner"
    cancelBtnText="test"
    confirmBtnText="Confirmar"
    customStyles={
      {
        "dateInput": {
          "borderWidth": 0,
        },
        "dateText": {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "700",
          "left": 0,
          "position": "absolute",
        },
        "dateTouchBody": {
          "width": "100%",
        },
        "disabled": {
          "backgroundColor": "transparent",
        },
        "placeholderText": {
          "color": "#fff",
          "fontSize": 18,
          "left": 0,
          "position": "absolute",
        },
      }
    }
    dark={false}
    date=""
    editable={false}
    format="DD/MM/YYYY"
    locale="pt-BR"
    mode="date"
    onDateChange={[Function]}
    placeholder=" "
    showIcon={false}
    style={
      [
        {
          "width": "100%",
        },
      ]
    }
  />,
  <View
    dark={false}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "height": 0.5,
        },
      ]
    }
  />,
]
`;

exports[`<DatePicker /> should render datepicker with custom props 1`] = `
[
  <Text
    accessibilityLabel=""
    collapsable={false}
    dark={false}
    error=""
    isPlaceholder={true}
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#ffffff",
          "lineHeight": 19,
          "paddingBottom": 16,
        },
        {
          "fontSize": 18,
          "top": 40,
        },
      ]
    }
    variant="md"
  />,
  <DatePicker
    androidMode="calendar"
    cancelBtnText="Cancelar"
    confirmBtnText="test"
    customStyles={
      {
        "dateInput": {
          "borderWidth": 0,
        },
        "dateText": {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "700",
          "left": 0,
          "position": "absolute",
        },
        "dateTouchBody": {
          "width": "100%",
        },
        "disabled": {
          "backgroundColor": "transparent",
        },
        "placeholderText": {
          "color": "#fff",
          "fontSize": 18,
          "left": 0,
          "position": "absolute",
        },
      }
    }
    dark={false}
    date=""
    editable={true}
    format="yyyy-MM-dd"
    locale="en"
    maxDate="2021-12-31"
    mode="date"
    onDateChange={[Function]}
    placeholder=" "
    showIcon={false}
    style={
      [
        {
          "width": "100%",
        },
      ]
    }
    testID="test"
  />,
  <View
    dark={false}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "height": 0.5,
        },
      ]
    }
  />,
]
`;

exports[`<DatePicker /> should render datepicker with error 1`] = `
[
  <Text
    accessibilityLabel=""
    collapsable={false}
    dark={false}
    error="test"
    isPlaceholder={true}
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#ffffff",
          "lineHeight": 19,
          "paddingBottom": 16,
        },
        {
          "fontSize": 18,
          "top": 40,
        },
      ]
    }
    variant="md"
  />,
  <DatePicker
    androidMode="spinner"
    cancelBtnText="Cancelar"
    confirmBtnText="Confirmar"
    customStyles={
      {
        "dateInput": {
          "borderColor": "#cc0000",
        },
        "dateText": {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "700",
          "left": 0,
          "position": "absolute",
        },
        "dateTouchBody": {
          "width": "100%",
        },
        "disabled": {
          "backgroundColor": "transparent",
        },
        "placeholderText": {
          "color": "#fff",
          "fontSize": 18,
          "left": 0,
          "position": "absolute",
        },
      }
    }
    dark={false}
    date=""
    editable={false}
    format="DD/MM/YYYY"
    locale="pt-BR"
    mode="date"
    onDateChange={[Function]}
    placeholder=" "
    showIcon={false}
    style={
      [
        {
          "width": "100%",
        },
      ]
    }
  />,
  <View
    dark={false}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "height": 0.5,
        },
      ]
    }
  />,
  <Text
    accessibilityLabel="Erro test"
    centered={false}
    large={false}
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 13,
          "lineHeight": 15.6,
        },
        {
          "color": "#FF4F3F",
          "marginTop": 4,
          "textAlign": "left",
        },
      ]
    }
    testID="error_testing"
    variant="xs"
  >
    test
  </Text>,
]
`;

exports[`<DatePicker /> should render datepicker with label 1`] = `
[
  <Text
    accessibilityLabel="test"
    collapsable={false}
    dark={false}
    error=""
    isPlaceholder={true}
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#ffffff",
          "lineHeight": 19,
          "paddingBottom": 16,
        },
        {
          "fontSize": 18,
          "top": 40,
        },
      ]
    }
    variant="md"
  >
    test
  </Text>,
  <DatePicker
    androidMode="spinner"
    cancelBtnText="Cancelar"
    confirmBtnText="Confirmar"
    customStyles={
      {
        "dateInput": {
          "borderWidth": 0,
        },
        "dateText": {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "700",
          "left": 0,
          "position": "absolute",
        },
        "dateTouchBody": {
          "width": "100%",
        },
        "disabled": {
          "backgroundColor": "transparent",
        },
        "placeholderText": {
          "color": "#fff",
          "fontSize": 18,
          "left": 0,
          "position": "absolute",
        },
      }
    }
    dark={false}
    date=""
    editable={false}
    format="DD/MM/YYYY"
    locale="pt-BR"
    mode="date"
    onDateChange={[Function]}
    placeholder=" "
    showIcon={false}
    style={
      [
        {
          "width": "100%",
        },
      ]
    }
  />,
  <View
    dark={false}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "height": 0.5,
        },
      ]
    }
  />,
]
`;

exports[`<DatePicker /> should render datepicker with value 1`] = `
[
  <Text
    accessibilityLabel=""
    collapsable={false}
    dark={false}
    error=""
    isPlaceholder={true}
    lineHeightVariant="min"
    style={
      [
        {
          "color": "#444444",
          "fontSize": 15,
          "lineHeight": 18,
        },
        {
          "color": "#ffffff",
          "lineHeight": 19,
          "paddingBottom": 16,
        },
        {
          "fontSize": 18,
          "top": 40,
        },
      ]
    }
    variant="md"
  />,
  <DatePicker
    androidMode="spinner"
    cancelBtnText="Cancelar"
    confirmBtnText="Confirmar"
    customStyles={
      {
        "dateInput": {
          "borderWidth": 0,
        },
        "dateText": {
          "color": "#fff",
          "fontSize": 14,
          "fontWeight": "700",
          "left": 0,
          "position": "absolute",
        },
        "dateTouchBody": {
          "width": "100%",
        },
        "disabled": {
          "backgroundColor": "transparent",
        },
        "placeholderText": {
          "color": "#fff",
          "fontSize": 18,
          "left": 0,
          "position": "absolute",
        },
      }
    }
    dark={false}
    date=""
    editable={false}
    format="DD/MM/YYYY"
    locale="pt-BR"
    mode="date"
    onDateChange={[Function]}
    placeholder=" "
    showIcon={false}
    style={
      [
        {
          "width": "100%",
        },
      ]
    }
  />,
  <View
    dark={false}
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "height": 0.5,
        },
      ]
    }
  />,
]
`;
